src/convertToObject.js

```diff
@@ -6,7 +6,20 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  let wasteStyles = sourceString.split(';');
+  const obj = {};
+
+  wasteStyles = wasteStyles.map((line) => line.trim());
+
+  for (const styles of wasteStyles.filter(Boolean)) {
+    const style = styles.split(':');
+    const key = style[0].trimStart().trimEnd();
+    const value = style[1].trimStart().trimEnd();
```

Comment:
Ensure that the `style` array has exactly two elements before accessing `style[0]` and `style[1]`. This will prevent potential runtime errors if the input string is malformed or if there are missing colons. You can add a check to handle such cases gracefully.